infrastructure/src/main/java/org/corfudb/infrastructure/InvokeCheckpointingJvm.java

```diff
@@ -0,0 +1,83 @@
+package org.corfudb.infrastructure;
+
+import org.corfudb.runtime.exceptions.NetworkException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+public class InvokeCheckpointingJvm implements IInvokeCheckpointing {
+
+    private static final long CONN_RETRY_DELAY_MILLISEC = 500;
```

Comment:
please use  Duration

Reply:
Since it's used for sleep, it might as well be long? 
As it needs to be converted to long at the usage.